Lib9c/Action/DailyReward.cs

```diff
@@ -51,77 +38,30 @@ public override IAccountStateDelta Execute(IActionContext context)
                 throw new FailedLoadStateException($"{addressesHex}Aborted as the game config was failed to load.");
             }
 
-            if (ctx.BlockIndex - avatarState.dailyRewardReceivedIndex >= gameConfigState.DailyRewardInterval)
+            if (avatarState.dailyRewardReceivedIndex < context.BlockIndex - gameConfigState.DailyRewardInterval)
             {
-                avatarState.dailyRewardReceivedIndex = ctx.BlockIndex;
-                avatarState.actionPoint = gameConfigState.ActionPointMax;
+                var sb = new StringBuilder()
+                    .Append($"{addressesHex}Not enough block index since the last received daily reward.")
+                    .Append(
+                        $" Expected: Equals or greater than {context.BlockIndex - gameConfigState.DailyRewardInterval}.")
+                    .Append($" Actual: {avatarState.dailyRewardReceivedIndex}");
+                throw new RequiredBlockIndexException(sb.ToString());
```

Comment:
I guess concatenate three strings directly isn't a big deal. 😅

Reply:
Habit is scary. hehe